sdk/go/x/auto/local_workspace.go

```diff
@@ -584,6 +584,14 @@ type GitRepo struct {
 	CommitHash string
 	// Optional function to execute after enlisting in the specified repo.
 	Setup SetupFn
+	// The absolute path to a private key for access to the github repo
```

Comment:
From a user-facing perspective, I could imagine it being confusing that there are four different top-level fields here that can be used in various combinations for auth. It's very clear from reading the examples in the godoc, but not necessarily from the IDE. I have two ideas on how we could make this more clear, and I'm curious to hear what you think:

1. We could enhance the comments on these fields to more clearly indicate the acceptable permutations (PAT, ssh + pass, user + pass).  It might also be helpful to include this in the top level comment for `GitRepo`
2. Another idea would be to hide this under an Auth field (in addition to adding some of the comments described in number 1):

```go
type GitRepo struct{
...
Auth GitAuth,
...
}

// GitAuth top-level comment indicating valid permutations and precedence.
type GitAuth struct {
        // The absolute path to a private key for access to the github repo
	SSHPrivateKeyPath string
	// The password that pairs with a username or as part of an SSH Private Key
	Password string
	// PersonalAccessToken is a GitHub personal access token is in replacment of your password
	PersonalAccessToken string
	// Username is the username to use when authenticating to a git repository
	Username string
}
```

Reply:
Agreed to help clear this up in the code and comments. Here's an example of how we handle similar logic in p/eks

checks: 

https://github.com/pulumi/pulumi-eks/blob/7998754242c87d2217b8a087494c83c74f451662/nodejs/eks/nodegroup.ts#L338-L351

---

type comments:

https://github.com/pulumi/pulumi-eks/blob/7998754242c87d2217b8a087494c83c74f451662/nodejs/eks/nodegroup.ts#L695-L711